invisible_cities/io/hits_io.py

```diff
@@ -21,26 +21,34 @@ def hits_from_df (dst : pd.DataFrame, skip_NN : bool = False) -> Dict[int, HitCo
     ------
     dst : pd.DataFrame
         DataFrame with obligatory columns :
-                event, time, npeak, nsipm, X, Y, Xrms, Yrms, Z,  Q, E
-        If Qc, Ec, track_id are not inside dst the default value is set to -1
+                event, npeak, X, Y, Z,  Q, E
+        If time, nsipm, Xrms, Yrms, Qc, Ec, track_id are not
+        inside dst the default value is set to -1
         If Xpeak, Ypeak not in dst the default value is -1000
     ------
     Returns
     ------
     Dictionary {event_number : HitCollection}
     """
     all_events = {}
-    for (event, time) , hits_df in dst.groupby(['event', 'time']):
+    for (event, time) , df in dst.groupby(['event',
+                                           getattr(dst,
+                                                   'time',
+                                                   [-1]*len(dst))]):
         #pandas is not consistent with numpy dtypes so we have to change it by hand
         event = np.int32(event)
         hits  = []
-        for i, row in hits_df.iterrows():
-            if skip_NN and row.Q == NN:
+        for i, row in df.iterrows():
+            if skip_NN and getattr(row,'Q', -1) == NN:
                 continue
             hit = Hit(row.npeak,
-                      Cluster(row.Q, xy(row.X, row.Y), xy(row.Xrms**2, row.Yrms**2),
-                              row.nsipm, row.Z, row.E,
-                              Qc = getattr(row, 'Qc', -1)),       # for backwards compatibility
+                      Cluster(getattr(row,'Q', row.E), xy(row.X, row.Y),
```

Comment:
This is becoming a bit silly, we are missing half of the variables to construct our `HitCollection` type...
It is also not true in case of Xrms and Yrms squared - here you are setting them to 0.

I am not saying you should deal with all event model problems in this PR but maybe doing something a bit cleaner like

```
for i, row in df.iterrows():
    Q = getattr(row,'Q', -1)
    if skip_NN and Q == NN:
        continue
    if hasattr(row, 'Xrms):
         Xrms = row.Xrms
         Xrms2 = Xrms**2
    else:
        Xrms =  Xrms2 = -1
    if hasattr(row, 'Yrms):
    ....
    nsipm = getattr...
    Qc = ...
```
This is still very ugly but avoids calling `getattr` for parameters that are used more than once, and it also makes it a bit more clear where we are setting this default -1 value

Reply:
Ok, I'll modify this to match your suggestion. It's completely true that this should be done just as a provisional solution: as you say, in the near future the **event model must be revisited and changed** in order to avoid these issues...

Reply:
Ok, thanks, I've changed the function trying to be a bit clearer. 
Yes, I obviously agree with you, it has no sense using this `HitCollection` here, however I'd say that that issue is not related with this PR (since it's meant to adapt the tracking algorithm to the current code), so **it should be mandatory to revisit the `event_model`in the near future.**